src/theory/strings/theory_strings_utils.cpp

```diff
@@ -44,6 +44,21 @@ Node mkAnd(std::vector<Node>& a)
   return NodeManager::currentNM()->mkNode(AND, au);
 }
 
+void getConjuncts(Node n, std::vector<Node>& conj)
```

Comment:
Sorry for being annoying but how about making this function take a `Kind` to flatten and making it non-recursive?

Reply:
Done.  Notice that this is pretty performance critical code, so I've made a special case for when the top symbol is not the kind, in which we don't do a traversal.